### PR TITLE
Translate heap allocation to Japanese

### PR DESCRIPTION
**Pull request by woodyZootopia**
TTSIA.

### CODE REVIEW

**Comment by phil-opp** [comment]
Thanks a lot! Is the translation ready for a review?

**Comment by woodyZootopia** [comment]
Sorry, I want to do some more stuffs and review by myself. I'll notify by changing this PR from draft to ready when it's done.

**Comment by phil-opp** [comment]
Ok sure, thanks!

**Comment by woodyZootopia** [comment]
The translation is finished.
I did my best to translate correctly, but if @JohnTitor or @garasubo have time, I would appreciate your review!

**Comment by JohnTitor** [comment]
I guess I have some time this weekend. Feel free to ignore me if I'm not around.

**Comment by woodyZootopia** [comment]
Hi @garasubo , are you interested in reviewing this PR? If so, please take your time. Otherwise, I think we can proceed?

**Review by garasubo — COMMENTED** [review]
Sorry for late response. I roughly reviewed your change, and it looks almost good. I added some suggestions. Thanks.

File: blog/content/edition-2/posts/10-heap-allocation/index.ja.md [hunk 1]
```diff
@@ -0,0 +1,806 @@
...
+}
+```
+
+([この例をplaygroundで実行する](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6186a0f3a54f468e1de8894996d12819))
+
+上の例の場合、参照を返すことには意味がありませんが、変数に関数よりも長く生存して欲しいというケースは存在します。すでに私たちのカーネルでそのようなケースに遭遇しています：それは[割り込み記述子表 (IDT) を読み込][load an interrupt descriptor table]もうとしたときで、ライフタイムを延ばすために`static`変数を使う必要があったのでした。
+
+[load an interrupt descriptor table]: @/edition-2/posts/05-cpu-exceptions/index.ja.md#idtwodu-miip-mu
+
+### スタティック変数
```
[the remark this answers is not in the document]
Comment (woodyZootopia) [reply blog/content/edition-2/posts/10-heap-allocation/index.ja.md]: ありがとうございます。統一性を持たせるため、ローカル変数も局所変数と直しました。

**Comment by woodyZootopia** [comment]
@garasubo Thank you for your review! I applied all of your suggestions (with very minor revisions). Please take a look if you want.

**Review by garasubo — APPROVED** [review]
LGTM

**Comment by woodyZootopia** [comment]
Thank you. Added @garasubo as the co-translator.
@phil-opp I believe this PR is ready to be merged!

**Pull request merged**